test/javascripts/lib/uploadHandler-test.js.es6

```diff
@@ -0,0 +1,30 @@
+import { fetchDataPromise } from "discourse/plugins/discourse-encrypt/lib/uploadHandler";
+
+QUnit.module("discourse-encrypt:lib:uploadHander");
+
+let testImageBase64 =
+  "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNkYPhfDwAChwGA60e6kgAAAABJRU5ErkJggg==";
+
+test("fetchDataPromise - image file", async assert => {
+  let uploadsUrl = {};
+  let file = new File([window.atob(testImageBase64)], "test.png", {
+    type: "image/png",
+    encoding: "utf-8"
+  });
+
+  // suppress the image onerror, it is not important
+  fetchDataPromise(file, uploadsUrl).catch(e => console.log(e));
```

Comment:
ESLint does not like console statements and we should keep the console clean too. :smile: 

```suggestion
  fetchDataPromise(file, uploadsUrl).catch(() => null);
```